app/data_models/progress_store.py

```diff
@@ -170,45 +152,39 @@ def update_section_or_repeating_blocks_progress_completion_status(
         return updated
 
     def get_section_or_repeating_blocks_progress_status(
-        self, section_id: str, list_item_id: str | None = None
+        self, section_key: SectionKey
     ) -> str:
         """
         Return the CompletionStatus of the Section or Repeating Blocks for a list item,
         specified by the given section_id and list_item_id.
         Returns NOT_STARTED if the progress does not exist
         """
-        progress_key = SectionKey(section_id, list_item_id)
-        if progress_key in self._section_and_repeating_blocks_progress:
-            return self._section_and_repeating_blocks_progress[progress_key].status
+        progress_key = section_key
+        if progress_key in self._progress:
+            return self._progress[progress_key].status
 
         return CompletionStatus.NOT_STARTED
 
-    def get_block_status(
-        self, *, block_id: str, section_id: str, list_item_id: str | None = None
-    ) -> str:
+    def get_block_status(self, *, block_id: str, section_key: SectionKey) -> str:
         """
         Return the completion status of the block specified by the given block_id,
         if it is part of the progress of the given Section or Repeating Blocks for list item
         specified by the given section_id or list_item_id
         """
-        blocks = self.get_completed_block_ids(
-            section_id=section_id, list_item_id=list_item_id
-        )
+        blocks = self.get_completed_block_ids(section_key)
         if block_id in blocks:
             return CompletionStatus.COMPLETED
 
         return CompletionStatus.NOT_STARTED
 
-    def get_completed_block_ids(
-        self, *, section_id: str, list_item_id: str | None = None
-    ) -> list[str]:
+    def get_completed_block_ids(self, section_key: SectionKey) -> list[str]:
         """
         Return the block ids recorded as part of the progress for the Section or Repeating Blocks
         for list item specified by the given section_id and list_item_id
         """
-        progress_key = SectionKey(section_id, list_item_id)
-        if progress_key in self._section_and_repeating_blocks_progress:
-            return self._section_and_repeating_blocks_progress[progress_key].block_ids
+        progress_key = section_key
```

Comment:
Should the instances of progess_key in this file now be section_key?

Reply:
Based on our conversations in tech sessions I think they are supposed to stay `progress keys`.

Reply:
Yeah wasn't sure just slightly confusing when there are lines like `progress_key = SectionKey(section_id, list_item_id)`